### PR TITLE
refactor(k8s): don't handle resource change detection based on error

### PR DESCRIPTION
**Pull request by lukidzi**
### Checklist prior to review

Refactor of [#11327](https://github.com/kumahq/kuma/pull/11327). We no longer need to handle this with an error because when the spec is not set, it gets compared with the same spec in `CreateOrUpdate` and doesn't update a resource.

- [ ] [Link to relevant issue][1] as well as docs and UI issues --
- [ ] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/plugins/runtime/k8s/controllers/pod_converter.go [hunk 1]
```diff
@@ -44,6 +44,7 @@ func (p *PodConverter) PodToDataplane(
 	services []*kube_core.Service,
 	others []*mesh_k8s.Dataplane,
 ) error {
+	logger := converterLog.WithValues("Dataplne.name", dataplane.Name, "Pod.name", pod.Name)
```
Comment (lahabana) [inline pkg/plugins/runtime/k8s/controllers/pod_converter.go:47]:
```suggestion
	logger := converterLog.WithValues("Dataplane.name", dataplane.Name, "Pod.name", pod.Name)
```

Comment (lahabana) [reply pkg/plugins/runtime/k8s/controllers/pod_converter.go:47]: It seems unusual to me to put `.` in log keys. @jakubdyszkiewicz opinions?

Comment (lukidzi) [reply pkg/plugins/runtime/k8s/controllers/pod_converter.go]: I wanted to preserve the same format as for ZoneIngress and ZoneEgress https://github.com/kumahq/kuma/blob/master/pkg/plugins/runtime/k8s/controllers/pod_converter.go#L66

**Review by Icarus9913 — APPROVED** [review]

File: pkg/plugins/runtime/k8s/controllers/pod_converter.go (hunk 1 above)
Comment (jakubdyszkiewicz) [reply pkg/plugins/runtime/k8s/controllers/pod_converter.go]: yeah, I'd try to avoid `.`

**Review by jakubdyszkiewicz — APPROVED** [review]

**Pull request merged**